resources/js/Pages/Adminland/Software/Edit.vue

```diff
@@ -0,0 +1,325 @@
+                      :label="$t('dashboard.expense_create_currency')"
+                      :custom-label-key="'code'"
+                      :custom-value-key="'id'"
+                      :placeholder="$t('dashboard.expense_create_currency')"
+                      :required="false"
+                      :value="form.currency"
+                      :datacy="'expense-currency'"
+                    />
+                  </div>
+                </div>
+              </div>
+            </div>
+
+            <!-- Purchase date -->
+            <div class="cf pa3 bb bb-gray pb4">
+              <div class="fl-ns w-third-ns w-100 mb3 mb0-ns">
+                <strong>{{ $t('account.software_new_purchased_date') }}</strong>
+                <p class="f7 silver lh-copy pr3-ns">
+                  {{ $t('employee.edit_information_timezone_help') }}
+                </p>
+              </div>
+              <div class="fl-ns w-two-thirds-ns w-100">
+                <div class="dt-ns dt--fixed di">
+                  <div class="dtc-ns pr2-ns pb0-ns w-100">
+                    <!-- year -->
+                    <text-input :id="'purchased_date_year'"
+                                v-model="form.purchased_date_year"
+                                :name="'purchased_date_year'"
+                                :errors="$page.props.errors.purchased_date_year"
+                                :label="$t('app.year')"
+                                :required="false"
+                                :type="'number'"
+                                :min="1900"
+                                :max="2050"
+                    />
+                  </div>
+                  <div class="dtc-ns pr2-ns pb0-ns w-100">
+                    <!-- month -->
+                    <text-input :id="'purchased_date_month'"
+                                v-model="form.purchased_date_month"
+                                :name="'purchased_date_month'"
+                                :errors="$page.props.errors.purchased_date_month"
+                                :label="$t('app.month')"
+                                :required="false"
+                                :type="'number'"
+                                :min="1"
+                                :max="12"
+                    />
+                  </div>
+                  <div class="dtc-ns pr2-ns pb0-ns w-100">
+                    <!-- day -->
+                    <text-input :id="'purchased_date_day'"
+                                v-model="form.purchased_date_day"
+                                :name="'purchased_date_day'"
+                                :errors="$page.props.errors.purchased_date_day"
+                                :label="$t('app.day')"
+                                :required="false"
+                                :type="'number'"
+                                :min="1"
+                                :max="31"
+                    />
```

Comment:
I'm wondering if we couldn't use a calendar component for that.
There a multiple cases when it could be more efficient, more easy to maintain, localisable, etc...

Reply:
I agree with your comment but in any case, it's out of scope of this PR.
But generally, yes we should have some kind of calendar. I just didn't want to put a huge library for that. If you find one, please tell me. And if we implement a library, there are many places in the app where we would need to update and completely change how dates are manipulated in the different services.